packages/backend/src/accounting/psql/service.ts

```diff
@@ -33,8 +33,12 @@ import {
   voidTransfers
 } from './ledger-transfer'
 import { LedgerTransfer, LedgerTransferType } from './ledger-transfer/model'
+import { TelemetryService } from '../../telemetry/service'
+import { RatesService } from '../../rates/service'
 
 export interface ServiceDependencies extends BaseService {
+  telemetry?: TelemetryService
+  aseRatesService?: RatesService
```

Comment:
That makes sense.

I'll have another look at the optionality of the services, but i'm pretty sure they ended up that way because telemetry is optional.
I did the optionality at the first possible injection layer, such that if enableTelemetry is not true, then the telemetry singleton service is not ever instantiated. (and as such would be undefined => ? on most uses.)

Reply:
> but i'm pretty sure they ended up that way because telemetry is optional.
Right, that makes sense. This makes me think whether we should support optional dependencies though. Might lead to some unexpected things. An approach could be just creating a noop telemetry service (with the same interface, but maybe just does console logs instead) and injecting that instead, when telemetry is disabled.

Reply:
>Might lead to some unexpected things

anything in particular?

I'm leaning a bit more into the optionality camp for now,
but I'm happy to change it and use a noop service if you think it's better.

Reply:
> anything in particular?

Mostly to avoid cases where you are trying to do `container.use(...)` expecting to get an instance of a service, but getting nothing/error instead. Edge casey - probably not a big deal for us. optional is ok

Reply:
Ok then, will leave it as optional for now 👍 🙇

Reply:
I think `aseRatesService` is unused here?

Reply:
it does, I just missed it. Thanks!
Regarding the noop, Is it ok if I do it after this gets in?

Reply:
> However, I wouldn't console.log but rather add proper logging to the noop service. 

Yup, agree.

> Is it ok if I do it after this gets in?

We can a follow up one to close this.